CMakeLists.txt

```diff
@@ -1,18 +1,21 @@
-if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-    set (SSL_ROOT_PATH ${CMAKE_CURRENT_LIST_DIR}/no-asm)
-else()
-    set (SSL_ROOT_PATH ${CMAKE_CURRENT_LIST_DIR})
-endif()
+function(android_add_openssl_to_targets)
```

Comment:
Since we're turning this into a cmake function, it would be more appropriate to turn this file into a *.cmake file like openssl.cmake, and then the user would include it as before and then call the function